cognite/neat/rules/exporter/_rules2dms.py

```diff
@@ -521,56 +498,186 @@ def find_existing_views(self, client: CogniteClient) -> set:
+
+        """
+
+        if components_to_create is None:
+            components_to_create = set(["all"])
+        existing_spaces = self.find_existing_spaces(client)
+        existing_containers = self.find_existing_containers(client)
+        existing_views = self.find_existing_views(client)
+        existing_data_model = self.find_existing_data_model(client)
+
+        if (
+            existing_spaces or existing_containers or existing_data_model or existing_views
+        ) and existing_component_handling == "fail":
+            raise exceptions.DataModelOrItsComponentsAlreadyExist(
+                existing_spaces, existing_data_model, existing_containers, existing_views
+            )
+
+        if "space" in components_to_create or "all" in components_to_create:
+            self.create_space(client, existing_spaces)
+        if "container" in components_to_create or "all" in components_to_create:
+            self.create_containers(client, existing_containers)
+        if "view" in components_to_create or "all" in components_to_create:
+            self.create_views(client, existing_views)
+        if "data model" in components_to_create or "all" in components_to_create:
+            self.create_data_model(client, existing_data_model)
+
+    def create_space(self, client: CogniteClient, existing_spaces: set | None = None):
         for space in self.spaces:
-            _ = client.data_modeling.spaces.apply(SpaceApply(space=space))
+            if not existing_spaces or space not in existing_spaces:
+                logging.info(f"Creating space {space} !!!")
+                _ = client.data_modeling.spaces.apply(SpaceApply(space=space))
+            else:
+                warnings.warn(
+                    f"Space {space} already exists in CDF! Skipping !!!",
+                    stacklevel=2,
+                )
+                logging.info(f"Space {space} already exists in CDF! Skipping !!!")
 
-    def create_containers(self, client: CogniteClient):
+    def create_containers(self, client: CogniteClient, existing_containers: set | None = None):
         for container_id in self.containers:
-            logging.info(f"Creating container {container_id} in space {self.space}")
-            _ = client.data_modeling.containers.apply(self.containers[container_id])
+            if not existing_containers or container_id not in existing_containers:
+                logging.info(f"Creating container {container_id}")
+                _ = client.data_modeling.containers.apply(self.containers[container_id])
+            else:
+                warnings.warn(
+                    f"Container {container_id} already exists! Skipping !!!",
+                    stacklevel=2,
+                )
+                logging.info(f"Container {container_id} already exists! Skipping !!!")
 
-    def create_views(self, client: CogniteClient):
+    def create_views(self, client: CogniteClient, existing_views: set | None = None):
```

Comment:
Same here, rewrite to one request. For views this is really important as they typically have dependencies between each other, and creating them one-at-a-time, will fail with 400 if they are out of order or have circular dependencies. All at once solves this up to 1000 views (after that you have to be more careful but that should be handled by the SDK and not neat)

Reply:
great ! done implemented